tests/unit/test_spatial_methods/test_scikit_finite_element_3d.py

```diff
@@ -0,0 +1,730 @@
+        grad_eval = grad_disc.evaluate(None, u_exact_num)
+        l2_error_r = np.sqrt(np.mean((grad_eval[:, 0] - grad_u_exact_r) ** 2))
+        l2_error_theta = np.sqrt(np.mean((grad_eval[:, 1] - grad_u_exact_theta) ** 2))
+        l2_error_z = np.sqrt(np.mean((grad_eval[:, 2] - grad_u_exact_z) ** 2))
+        assert l2_error_r < 0.1
+        assert l2_error_theta < 5e-2
+        assert l2_error_z < 0.1
+
+    def test_laplacian_cylinder_manufactured(self):
+        mesh = get_unit_3d_mesh_for_testing(
+            geom_type="cylinder", r_inner=0.2, radius=0.5, height=1.0, h=0.1
+        )
+        model = pybamm.BaseModel()
+        u = pybamm.Variable("u", "current collector")
+        r = pybamm.SpatialVariable("r", "current collector")
+        z = pybamm.SpatialVariable("z", "current collector")
+        u_exact = r**2 * z
+        f = -4 * z
+        model.algebraic = {u: pybamm.laplacian(u) - pybamm.source(f, u)}
+        model.boundary_conditions = {
+            u: {
+                "bottom": (u_exact, "Dirichlet"),
+                "top": (u_exact, "Dirichlet"),
+                "inner radius": (u_exact, "Dirichlet"),
+                "outer radius": (u_exact, "Dirichlet"),
+            }
+        }
+        model.initial_conditions = {u: pybamm.Scalar(0)}
+        model.variables = {"u": u}
+        spatial_methods = {"current collector": pybamm.ScikitFiniteElement3D()}
+        disc = pybamm.Discretisation(mesh, spatial_methods)
+        disc.process_model(model)
+        solver = pybamm.AlgebraicSolver()
+        solution = solver.solve(model)
+        u_numerical = solution.y.flatten()
+        nodes = mesh["current collector"].nodes
+        x_nodes, y_nodes, z_nodes = nodes.T
+        r_nodes = np.sqrt(x_nodes**2 + y_nodes**2)
+        u_analytical = r_nodes**2 * z_nodes
+        l2_error = np.sqrt(np.mean((u_numerical - u_analytical) ** 2))
+        assert l2_error < 0.08
+
+    def test_divergence_cylinder_manufactured(self):
+        mesh = get_unit_3d_mesh_for_testing(
+            geom_type="cylinder", r_inner=0.1, radius=0.5, height=1.0, h=0.15
+        )
+        disc = pybamm.Discretisation(
+            mesh, {"current collector": pybamm.ScikitFiniteElement3D()}
+        )
+        u = pybamm.Variable("u", domain="current collector")
+        disc.set_variable_slices([u])
+
+        nodes = mesh["current collector"].nodes
+        x, y, z = nodes.T
+        r = np.sqrt(x**2 + y**2)
+
+        # Manufactured solution u = r^3 * z
+        # div(grad(u)) = lap(u) = 9*r*z
+        u_exact_num = r**3 * z
+        div_grad_u_exact_num = 9 * r * z
+
+        div_grad_u_sym = pybamm.div(pybamm.grad(u))
+        div_grad_u_disc = disc.process_symbol(div_grad_u_sym)
+        div_grad_u_eval = div_grad_u_disc.evaluate(None, u_exact_num).flatten()
+
+        l2_error = np.sqrt(np.mean((div_grad_u_eval - div_grad_u_exact_num) ** 2))
+        assert l2_error < 1.5
+
+    def test_cylinder_mesh_properties(self):
```

Comment:
can you add some tests for disctretising known expressions (e.g. y = r * theta + z) and equations?